### ISSUE TITLE
[popover] accessibility tab documentation

### ISSUE THREAD

**Issue by laurenmrice** [carbon-design-system/carbon-website#2475]
## Accessibility tab
- Add accessibility documentation for the Popover component. Use our current accessibility docs template.

Possible sections:
- [ ] How it works 
- [ ] Accessibility considerations 
- [ ] Resources
- [ ] Accessibility testing
- [ ] Automated test
- [ ] macOS Screen reader tests
- [ ] Windows screen reader tests
- [ ] Android screen reader tests
- [ ] iOS screen reader tests

**Comment by joshblack** [comment]
When tackling this, it seems like this content was more appropriate in other pages (like tooltip or disclosure specifically) as there wasn't much related to popover specifically. Going to close this out as a result 👀
